pilot/prompts/development/task/update_task.prompt

```diff
@@ -1,37 +1,46 @@
 {% if completed_steps -%}
-# Completed Task Steps:
-```
-{{ completed_steps }}
+Here is a list of completed steps:
 ```
+{% for step in completed_steps %}
+step #{{ loop.index }}:
+step_type: {{ step.type }}{% if 'command' in step -%}
+command_to_run: `{{ step.command.command }}`{% elif step.type == 'code_change' %}
+code_change_description: `{{ step.code_change_description }}`{% endif -%}
 
+{% endfor %}
+```
 {% endif -%}
 
-# Current Step:
-{% if success %}This step was executed successfully.
-{%- elif user_input -%}
-This step will not be executed. {{ user_input }}
-{%- else -%}
-This step was not executed successfully.
-{%- endif %}
+Right now, we're working on the step #{{ current_step_index }}:
 ```
-{{ current_step }}
+step_type: {{ current_step.type }}
+command_to_run: `{{ current_step.command.command }}`
 ```
 
-{%- if cli_response %}
-
+We've run the command `{{ current_step.command.command }}` and we got the following output:
 {{ cli_response }}
-{%- endif %}
 
 {%- if next_steps %}
 
 # Next Task Steps:
 ```
-{{ next_steps }}
+{% for step in next_steps %}
+step #{{ loop.index + current_step_index + 1 }}:
+step_type: {{ step.type }}{% if 'command' in step -%}
+command_to_run: `{{ step.command.command }}`{% endif -%}
+
+{% endfor %}
 ```
 
-Based on the Next Task Steps and my comments above for the Current Step, please create a new list of task steps to complete this task. The Current Step will need to be modified, and based on that any of the Next Task Steps may or may not still be valid.
-{% else -%}
-Based on my comments above for the Current Step, please create a new list of task steps to complete this task.
+Now, think step by step and create a new list of task steps to complete this task based on the output we got from running `{{ current_step.command.command }}`. It will either be the same as it is now or any of the Next Task Steps may or may not be needed anymore.
```

Comment:
do we always have access to current_step.command.command?

Reply:
We should have. This prompt is being used only when the command is being run while for code change or human intervention this won't be called at all - these steps will just be executed. Likely, in the future, this might change but will leave it to fail then so we can fix it appropriately and not put an if statement around it now because then, it will be harder to find the bug